src/Discord.Net.Core/Entities/Messages/MessageReference.cs

```diff
@@ -0,0 +1,33 @@
+using System.Diagnostics;
+
+namespace Discord
+{
+    /// <summary>
+    ///     Contains the IDs sent from a crossposted message.
+    /// </summary>
+    [DebuggerDisplay(@"{DebuggerDisplay,nq}")]
+    public class MessageReference
+    {
+        /// <summary>
+        ///     Gets the Message ID of the original message.
+        /// </summary>
+        public ulong? MessageId { get; internal set; }
```

Comment:
This is actually optional, not nullable. The type on the docs would be `snowflake?` if nullable. 

Should be `Optional<ulong>`.